ckanext/dgu/bin/set_missing_resource_formats.py

```diff
@@ -0,0 +1,113 @@
+"""
+For those resources that do not have valid formats (i.e. empty, or just whitespace), this
+script will set the value from the QA object that was previously created.
+
+This will trigger an archive/packagezip/qa cycle, so should not be used too frequently.
+"""
+from sqlalchemy.exc import IntegrityError
+from pprint import pprint
+import ckan.plugins.toolkit as t
+from optparse import OptionParser
+
+from ckan.logic import ValidationError
+from ckanext.dgu.bin.common import get_ckanapi
+from running_stats import Stats
+
+stats = Stats()
+ds_stats = Stats()
+
+class SetResourceFormatsCommand(object):
+
+    def __init__(self, config_or_url):
+        self.ckan = get_ckanapi(config_or_url)
+
+    def update_resource_dict(self, resource):
+        """ Set the format on the resource to the format determined by QA """
+        try:
+            qa_info = self.ckan.action.qa_resource_show(id=resource['id'])
+            if qa_info.get('format', ''):
+                resource['format'] = qa_info['format']
+                stats.add("Updating format to %s" % qa_info['format'], resource['id'])
```

Comment:
perhaps you need to cope with the situation where `qa_info.get('format', '')` is not True? It should probably return False. And I imagine it would be usefully logged in the stats.